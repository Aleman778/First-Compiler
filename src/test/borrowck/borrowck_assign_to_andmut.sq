// ignore

fn a(s: &i32) {
    *s = *s + 1;
}

fn b(s: &mut i32) {
    *s = *s + 1;
}

fn c(s: & &mut i32) {
    **s = **s + 1;
}

fn main() {}
