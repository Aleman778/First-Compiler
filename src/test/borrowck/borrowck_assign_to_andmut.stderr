error: cannot assign to `*s` which is behind a `&` reference
  --> $DIR/borrowck_assign_to_andmut.sq:4:5
   |
LL | fn a(s: &i32) {
   |         ---- help: consider changing this to be a mutable reference: `&mut i32`
LL |     *s = *s + 1;
   |     ^^^^^^^^^^^ `s` is a `&` reference, so the data it refers to cannot be written

error: cannot assign to `**s` which is behind a `&` reference
  --> $DIR/borrowck_assign_to_andmut.sq:12:5
   |
LL | fn c(s: & &mut i32) {
   |         ---------- help: consider changing this to be a mutable reference: `&mut &mut i32`
LL |     **s = **s + 1;
   |     ^^^^^^^^^^^^^ `s` is a `&` reference, so the data it refers to cannot be written